armory/data/datasets.py

```diff
@@ -20,70 +21,98 @@
 logger = logging.getLogger(__name__)
 
 
-def _in_memory_dataset_tfds(
-    dataset_name: str, dataset_dir: str, preprocessing_fn: Callable,
-) -> (np.ndarray, np.ndarray, np.ndarray, np.ndarray):
-    default_graph = tf.compat.v1.keras.backend.get_session().graph
+class ArmoryDataGenerator(DataGenerator):
+    def __init__(self, generator, size, batch_size, preprocessing_fn=None):
+        super().__init__(size, batch_size)
+        self.preprocessing_fn = preprocessing_fn
+        self.generator = generator
+        self.total_iterations = size // batch_size
```

Comment:
Do we know how tfds handles partial batches at the end of the epoch/dataset?

Reply:
Yeah remainders would have been returned. Updated code PTAL

Reply:
Reverted to make the remainders still returned (needed for testing and shouldn't affect training).  Kept the new method of batching though as its more configurable in the future.